swiftide-indexing/src/pipeline.rs

```diff
@@ -170,6 +180,32 @@ impl Pipeline {
     #[must_use]
     pub fn filter_cached(mut self, cache: impl NodeCache + 'static) -> Self {
         let cache = Arc::new(cache);
+        let (cache_sender, mut cache_receiver) = mpsc::channel(1000);
+
+        let cache_for_task = Arc::clone(&cache);
+
+        let handle = tokio::spawn(async move {
+            while let Some(cache_key) = cache_receiver.recv().await {
+                let cache_node = Node {
+                    parent_id: Some(cache_key),
+                    ..Default::default()
+                };
+                cache_for_task.set(&cache_node).await;
+            }
+        });
+
+        self.cache_handle = Some(
+            handle
+                .map(|result| {
+                    if let Err(err) = result {
+                        // TOCHECK: Should we continue or panic here?
+                        tracing::error!("Cache task failed: {err}");
```

Comment:
I think logging is enough? I'd rather avoid panics if possible. Alternative would be to propagate the error back to the main stream